Benjamin Bierlein 76418644 Econ 125 HW 3 Spring 2025

In [ ]:
##Import necessary packages
import numpy as np  # Useful for math calculations
import pandas as pd  # Useful for data manipulation
import statsmodels.api as sm  # Useful for regression analysis
import statsmodels.formula.api as smf  # Required for regression using formulas
import matplotlib.pyplot as plt  # Useful for plotting

##Load Data
ct_df = pd.read_csv("/content/collegetown.csv")
ct_df.head()
print(ct_df.head())

##Interaction Term
ct_df["close_ln_sqft"] = ct_df["close"] * np.log(ct_df["sqft"])

##Add log(price) and log(sqft) for regression
ct_df["ln_price"] = np.log(ct_df["price"])
ct_df["ln_sqft"] = np.log(ct_df["sqft"])

##Estimate the model
model1 = smf.ols("ln_price ~ ln_sqft + close + close_ln_sqft", data=ct_df).fit()

# Step 4: Summary of the model
model1_summary = model1.summary()
model1_summary



   price   sqft  age  pool  fireplace  close  twostory  occupied
0   96.5  16.77   10     0          0      0         0         0
1  106.0  17.51   10     0          0      0         0         0
2  114.0  18.37    9     0          0      0         0         1
3  119.9  29.10    9     0          0      0         0         0
4  122.7  14.68    9     0          0      0         0         1


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               ln_price   R-squared:                       0.542
Model:                            OLS   Adj. R-squared:                  0.539
Method:                 Least Squares   F-statistic:                     195.4
Date:                Sat, 03 May 2025   Prob (F-statistic):           1.26e-83
Time:                        01:52:12   Log-Likelihood:                -169.13
No. Observations:                 500   AIC:                             346.3
Df Residuals:                     496   BIC:                             363.1
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         1.2714      0.186      6.822      0.000       0.905       1.638
ln_sqft           1.2410      0.057     21.819      0.000       1.129       1.353
close             1.9957      0.306      6.515      0.000       1.394       2.598
close_ln_sqft    -0.5560      0.094     -5.910      0.000      -0.741      -0.371
==============================================================================
Omnibus:                       15.465   Durbin-Watson:                   0.898
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               26.764
Skew:                          -0.191   Prob(JB):                     1.54e-06
Kurtosis:                       4.067   Cond. No.                         84.1
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

close: this variable is statistically significant with high t value and p = 0.00 < .05

close_ln_sqf: this variable is statistically significant with high negative t value and p = 0.00 < .05

In [ ]:
##Predict prices for 2500 sqft homes (close and not close)
ln_sqft_2500 = np.log(2500)

##Predict for house NOT close to university (close = 0)
ln_price_not_close = model1.params['Intercept'] + model1.params['ln_sqft'] * ln_sqft_2500

##Predict for house CLOSE to university (close = 1)
ln_price_close = (
    model1.params['Intercept']
    + model1.params['ln_sqft'] * ln_sqft_2500
    + model1.params['close']
    + model1.params['close_ln_sqft'] * ln_sqft_2500
)
## Print predicted prices
print(f"Predicted price for 2500 sqft house NOT close to university: ${price_not_close:.2f}")
print(f"Predicted price for 2500 sqft house CLOSE to university:     ${price_close:.2f}")

##Convert back from log price to actual price
price_not_close = np.exp(ln_price_not_close)
price_close = np.exp(ln_price_close)

##Estimate new model including FIREPLACE, TWOSTORY, OCCUPIED
model2 = smf.ols("ln_price ~ ln_sqft + close + close_ln_sqft + fireplace + twostory + occupied", data=ct_df).fit()

model2.summary()

Predicted price for 2500 sqft house NOT close to university: $58771.83
Predicted price for 2500 sqft house CLOSE to university:     $5580.40


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               ln_price   R-squared:                       0.578
Model:                            OLS   Adj. R-squared:                  0.573
Method:                 Least Squares   F-statistic:                     112.7
Date:                Sat, 03 May 2025   Prob (F-statistic):           3.87e-89
Time:                        02:03:35   Log-Likelihood:                -148.30
No. Observations:                 500   AIC:                             310.6
Df Residuals:                     493   BIC:                             340.1
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         1.5913      0.197      8.075      0.000       1.204       1.979
ln_sqft           1.0983      0.063     17.564      0.000       0.975       1.221
close             1.8603      0.296      6.280      0.000       1.278       2.442
close_ln_sqft    -0.5222      0.091     -5.747      0.000      -0.701      -0.344
fireplace         0.0909      0.033      2.786      0.006       0.027       0.155
twostory          0.1015      0.047      2.165      0.031       0.009       0.194
occupied          0.1608      0.030      5.364      0.000       0.102       0.220
==============================================================================
Omnibus:                       10.167   Durbin-Watson:                   0.982
Prob(Omnibus):                  0.006   Jarque-Bera (JB):               16.878
Skew:                          -0.078   Prob(JB):                     0.000216
Kurtosis:                       3.886   Cond. No.                         87.6
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Predicted price for 2500 sqft house NOT close to university: $58771.83

Predicted price for 2500 sqft house CLOSE to university:     $5580.40

A house with a fireplace is associated with ~9.1% higher price, all else equal.

A two-story house is associated with ~10.2% higher price, all else equal.

An occupied house is associated with ~16.1% higher price, all else equal.